### PR TITLE
Add unit tests and linting config

### PR DESCRIPTION
**Pull request by aersida**
This includes fixing most of the eslint reported issues

### CODE REVIEW

File: gettext.js [hunk 1]
```diff
@@ -51,15 +57,17 @@ if (typeof Object.assign != 'function') {
     };
 
     /** Convert any markdown in the sourceText, stripping off any bracketing HTML tags */
-    global.markdownText = function (sourceText) {
-        var conversion = mdConverter.makeHtml(sourceText).trim();
+    global.markdownText = function(sourceText) {
+        // eslint-disable-next-line no-undef
+        global.mdConverter = global.mdConverter || new showdown.Converter();
```
Comment (raphaelmerx) [inline gettext.js]: `mdConverter` doesn't need to be a global variable, it's only used locally.

File: test/gettext.spec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/* global beforeEach describe expect gettext it setLanguage */
+/* eslint-env mocha node */
+
+// gettext.js is included in the karma.conf.js
+
+describe('gettext', () => {
```
Comment (raphaelmerx) [inline test/gettext.spec.js]: thanks for the thorough tests!

**Review by raphaelmerx — APPROVED** [review]
Looks good!

File: gettext.js (hunk 1 above)
Comment (md8n) [reply gettext.js]: I remember I made it global so that downstream unit tests would work properly.

**Pull request merged**